topic7/FileManager.java

```diff
@@ -0,0 +1,46 @@
+package topic7;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+public class FileManager {
+
+    static void scriereaInformatieTextInFisier(String caleaCatreFisier, String date) throws IOException {
+        File fisier = new File(caleaCatreFisier);
+        try {
+            if (fisier.createNewFile()) {
+                System.out.println("File created: " + fisier.getName());
+            } else {
+                System.out.println("File already exists!");
+            }
+
+            FileWriter fileWriter = new FileWriter(caleaCatreFisier);
+            fileWriter.write(date);
+            fileWriter.close();
+
+        } catch (IOException e) {
+            System.out.println("An error occurred: " + e.getMessage());
+        }
+    }
+
+    static void alipireaInformatieiLaFisier(String caleaCatreFisier, String date) throws IOException {
+        try (FileWriter fileWriter = new FileWriter(caleaCatreFisier, true)) {
+            fileWriter.append("\n");
+            fileWriter.append(date);
+        } catch (IOException e) {
```

Comment:
please close fileWriter

Reply:
am pus initial fileWriter.close() dar arata redundant din cauza ca creez fileWriter-ul in cadrul parantezelor de la try, ceeea ce automat inseamna ca la se inchide cind se termina blocul try.

Asa e si in python, daca folosesti with la deschiderea file-urilor pai nu mai trebuie sa inchizi file-ul, el automat e handluit